src/future/future.ts

```diff
@@ -78,10 +78,9 @@ class Future<A> {
         this.promise
           .then((value: A): void => {
             try {
-              mapper(value)
-                .foreach((b: B): void => {
-                  resolve(b);
-                });
+              mapper(value).promise
```

Comment:
I think rejection should be handled here. Otherwise, the future will never be completed when map function produces a Future that wraps rejected Promise.

In Scala, Following code results in failed Future.

```
> Future(1).flatMap { v => Future.failed(new Exception("foo")) }
Future(Failure(java.lang.Exception: foo))
```

What do you think about that?